mycore-base/src/main/java/org/mycore/access/facts/condition/combined/MCRAbstractCombinedCondition.java

```diff
@@ -64,4 +68,14 @@ public void debugInfoForMatchingChildElement(MCRCondition c, boolean matches) {
         }
     }
 
+    boolean addDebugInfoIfRequested(MCRCondition c, MCRFactsHolder facts) {
+        if (!isDebug()) {
+            return c.matches(facts);
+        }
+        boolean matches = c.matches(facts);
+        Objects.requireNonNull(c.getBoundElement(), "Condition is not bound to an element.")
```

Comment:
Is `Objects#requireNonNull` really necessary here. This introduces a new possibility for an exception that can only show itself when debug is enabled. The old implementation seems more robust.